src/mono/mono/mini/aot-compiler.c

```diff
@@ -13302,19 +13302,17 @@ add_mibc_group_method_methods (MonoAotCompile *acfg, MonoMethod *mibcGroupMethod
 
 		g_assert (cur + 4 < end); // Assert that there is atleast a 32 bit token before the end
 		guint32 mibcGroupMethodEntryToken = read32 (cur + 1);
-		g_assertf ((mono_metadata_token_table (mibcGroupMethodEntryToken) == MONO_TABLE_MEMBERREF || mono_metadata_token_table (mibcGroupMethodEntryToken) == MONO_TABLE_METHODSPEC), "token %x is not MemberRef or MethodSpec.\n", mibcGroupMethodEntryToken);
+		g_assertf ((mono_metadata_token_table (mibcGroupMethodEntryToken) == MONO_TABLE_MEMBERREF || mono_metadata_token_table (mibcGroupMethodEntryToken) == MONO_TABLE_METHODSPEC), "token '%x' is not MemberRef or MethodSpec.\n", mibcGroupMethodEntryToken);
 		cur += op_size;
 
-		MonoMethod *methodEntry = mono_get_method_checked (image, mibcGroupMethodEntryToken, mibcModuleClass, context, error);
-		mono_error_assert_ok (error);
-
-		MonoClass *method_class = mono_method_get_class (methodEntry);
-		if (!method_class)
-			continue;
-
-		MonoImage *method_image = mono_class_get_image (method_class);
-		if (!method_image)
+		// Skip methods in the profile that are not found
+		ERROR_DECL(method_entry_error);
+		MonoMethod *methodEntry = mono_get_method_checked (image, mibcGroupMethodEntryToken, mibcModuleClass, context, method_entry_error);
+		if (!is_ok (method_entry_error)) {
+			g_warning ("Method not found for method token '%x' from mibc image '%s'.\n", mibcGroupMethodEntryToken, image->name);
```

Comment:
Will this end up as an MSBuild warning? I think we should log it, as long as it doesn't show up as a warning for users.

Reply:
I don't think it should make it through as an MSBuild warning. I tested locally and I could only get the warning to pop up when running the mono aot compiler manually. `dotnet publish` doesn't surface the log. I've been told by @akoeplinger that through a certain [format](https://discord.com/channels/732297728826277939/737723183397994563/1002607657926017157), it would end up as a MSBuild warning, but I believe this message shouldn't.

Reply:
The profile not matching the app is normal behavior so I don't think this should cause a warning.

Reply:
Is there a scenario where a profile not matching the app is problematic and should be noted? What countermeasure do/should we have for this? I originally encountered an unfound method when accidentally profiling an app that was untrimmed and then used the corresponding .mibc to Profiled AOT a trimmed version. Though I suppose thats not problematic as the methods in the trimmed version should be in the profile anyways?

Reply:
It might be useful to have some kind of functionality to show the user when the profile doesn't match the app any more, but it should be done in some kind of user friendly way not a g_warning ().